File: maximum_nesting_depth_of_the_parentheses.go
func maxDepth(s string) int {
    curr, ans := 0, 0
    
    for _, ch := range s {
        if ch == '(' {
            curr++
        } else if ch == ')' {
            curr--
        }

        ans = max(curr, ans)
    }
    return ans
}

// Time Complexity : O(n)
// Space Complexity : O(1)

File: maximum_nesting_depth_of_the_parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int curr = 0, ans = 0;

        for(char ch: s) {
            if(ch == '(') {
                curr++;
            } else if(ch == ')') {
                curr--;
            }

            ans = max(curr, ans);
        }
        return ans;
    }
};

// Time Complexity: O(n)
// Space Complexity: O(1)
